catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -806,46 +811,62 @@ private String guessMimeType(String mimeTypeRaw, String fileName,
 
     private void generateMetacardAndContentItems(StorageRequest storageRequest,
             List<ContentItem> incomingContentItems, Map<String, Metacard> metacardMap,
-            List<ContentItem> contentItems, List<FileBackedOutputStream> fileBackedOutputStreams)
+            List<ContentItem> contentItems, Map<String, Path> tmpContentPaths)
             throws IngestException {
         for (ContentItem contentItem : incomingContentItems) {
             try {
-                FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(1000000);
-                fileBackedOutputStreams.add(fileBackedOutputStream);
+                Path tmpPath = null;
                 long size;
                 try {
                     if (contentItem.getInputStream() != null) {
-                        size = IOUtils.copyLarge(contentItem.getInputStream(),
-                                fileBackedOutputStream);
+                        tmpPath =
+                                Files.createTempFile(FilenameUtils.getBaseName(contentItem.getFilename()),
+                                        FilenameUtils.getExtension(contentItem.getFilename()));
```

Comment:
It looks like this drops all of the temp files in the same directory. We probably need to prepend/append the content item ID so that we don't clobber files that just happen to have the same name.